README.md

```diff
@@ -35,6 +35,9 @@ End`;
 
 const EXPECTED_SOLUTION = {
   IsLinear: true,
+  IsQuadratic: false,
+  Status: 'Optimal',
+  ObjectiveValue: 125.2083333333333,
```

Comment:
this will break the tests if the optimizer starts returning any value that is not strictly equal. This is dangerous, since highs gives no guarantee on the precision of results

Reply:
Hm, no, probably not directly but I guess it does allow you to control the precisions involved to some extent (cf. e.g. primal_feasibility_tolerance).

If it's a problem here, wouldn't it also be a problem for the decision variables themselves? For instance, with

```
Maximize
  x1
Subject to
  24 x1 <= 3005
End
```

the result becomes

```
{
  "Status": "Optimal",
  "Columns": {
    "x1": {
      "Index": 0,
      "Status": "BS",
      "Lower": 0,
      "Upper": null,
      "Primal": 125.208,
      "Dual": 0,
      "Name": "x1"
    }
  },
  "Rows": [
    {
      "Index": 0,
      "Status": "UB",
      "Lower": null,
      "Upper": 3005,
      "Primal": 3005,
      "Dual": 1
    }
  ],
  "IsLinear": true,
  "IsQuadratic": false
}
```

(As an aside, I'm surprised the precision isn't higher on this one. Maybe the pretty printer is responsible for unnecessarily lowering the precision?)

Are there situations where you would actually _like_ the test to fail, so as to be able to be alerted when a given update of HiGHS causes a change in behavior?

All of that aside, could a potential fix be to change the test itself to one whose objective at an optimal value is a representable number with a finite (and not too long) decimal expansion?

Reply:
> I'm surprised the precision isn't higher on this one. Maybe the pretty printer is responsible for unnecessarily lowering the precision? Edit: Indeed, it shows `x1 125.2083333333333` in the solution file.

That's true for the objective value with the pretty printer, because it's just putting out a single value

However, primal and dual values are only reported to about 6sf with the pretty printer so, in this case, it gives `x1=125.208`. The "raw" solution output gives full precision in the real values.

Reply:
> > I'm surprised the precision isn't higher on this one. Maybe the pretty printer is responsible for unnecessarily lowering the precision? Edit: Indeed, it shows `x1 125.2083333333333` in the solution file.
> 
> That's true for the objective value with the pretty printer, because it's just putting out a single value
> 
> However, primal and dual values are only reported to about 6sf with the pretty printer so, in this case, it gives `x1=125.208`. The "raw" solution output gives full precision in the real values.

Hello @jajhall ,

Then please let me expand on this.
As the JavaScript interface is done through the pretty printer, would it be possible to increase the pretty printer precision? Or have it configurable? 

Cheers,

Roman

Reply:
I'll increase the precision of the objective value in the pretty solution file, but there's no case for having it configurable. 

If you want "full precision" results, use the raw solution file - that's what it's for. The pretty and raw formats exist for very different reasons.

Reply:
Hmm, let me add some precisions.

Re. above, you wrote that _x1=125.208_ is actually not the full precision, but that the "raw" output file contains full precision.

So, in this case, my point would be that the JavaScript wrapper of HiGHS should rather parse, **and return as output of the .solve() function**, the values contained in the raw solution file, and not the ones of the pretty solution file.

So, in the @fuglede example above, we would have _x1=125.207326487463834_ (whatever :-)).

Reply:
> So, in this case, my point would be that the JavaScript wrapper of HiGHS should rather parse, **and return as output of the .solve() function**, the values contained in the raw solution file, and not the ones of the pretty solution file.

Ideally, yes, but there may only have been the pretty solution output when the JavaScript wrapper of HiGHS was written, or the old "raw" format that wasn't as easy to parse as the current format - still easier than the pretty format. Whatever, once the JavaScript wrapper of HiGHS was based on the pretty format and "worked" I understand why it was not felt worth doing the work to make the switch.

Reply:
Ah, undestood.

The weight of history :-) !

I'll open a feature request to track this one then.

I think better precision in output will benefit all users.

Reply:
Updated the test and README to only use representable numbers.